sycl/source/detail/device_image_impl.hpp

```diff
@@ -270,10 +274,15 @@ class device_image_impl {
       //  TODO consider changing the lifetime of device_image_impl instead
       ur_buffer_properties_t Properties = {UR_STRUCTURE_TYPE_BUFFER_PROPERTIES,
                                            nullptr, MSpecConstsBlob.data()};
-      memBufferCreateHelper(
-          Plugin, detail::getSyclObjImpl(MContext)->getHandleRef(),
-          UR_MEM_FLAG_READ_WRITE | UR_MEM_FLAG_ALLOC_COPY_HOST_POINTER,
-          MSpecConstsBlob.size(), &MSpecConstsBuffer, &Properties);
+      try {
+        memBufferCreateHelper(
+            Plugin, detail::getSyclObjImpl(MContext)->getHandleRef(),
+            UR_MEM_FLAG_READ_WRITE | UR_MEM_FLAG_ALLOC_COPY_HOST_POINTER,
+            MSpecConstsBlob.size(), &MSpecConstsBuffer, &Properties);
+      } catch (std::exception &e) {
+        __SYCL_REPORT_EXCEPTION_TO_STREAM(
+            "exception in get_spec_const_buffer_ref", e);
```

Comment:
This will be a user-visible message, so it should have more information

Reply:
The exception details comes from the origin (far down the call chain).
This function is to catch it and exit gracefully with whatever propagated from the exception origin.
There is no way at this outermost level can specify the details of the exception.